engine/graphics/src/webgpu/graphics_webgpu.cpp

```diff
@@ -354,7 +385,11 @@ static void WebGPUSetTextureInternal(WebGPUTexture *texture, const TextureParams
             const size_t dataSize = bpp * params.m_Width * params.m_Height * depth;
 
             dest.texture = texture->m_Texture;
-            layout.bytesPerRow = extent.width * bpp;
+            layout.bytesPerRow = extent.width;
+            if(IsTextureFormatCompressed(params.m_Format))
+                layout.bytesPerRow = (layout.bytesPerRow / WebGPUCompressedBlockWidth(params.m_Format)) * WebGPUCompressedBlockByteSize(params.m_Format);
```

Comment:
If `params.m_Format == dmGraphics::TEXTURE_FORMAT_RGBA_PVRTC_4BPPV1` we get divizion-by-zero here

Reply:
Only ASTC and BC are ever supported in current WebGPU port. In theory could do ETC but I have not tested it, thus never advertise support.